### ISSUE TITLE
App Re-initialise Priority

### ISSUE THREAD

**Issue by Odianosen25** [home-assistant/appdaemon#490]
It was noticed that though when AD restarts normally, the priority config on apps are followed, and the apps come up fine as expected in the right other.
But when a plugin restarts, which leads to the termination and initialisation of apps, this priority config is not followed
It leads to some apps, that shouldn't start up before others based on priority settings, starting up first and things can be naughty sometimes.

It seems to simply follow the reverse other in which the apps were terminated.

Regards

**Comment by Odianosen25** [comment]
This is a recent log
```
2019-01-22 20:00:47.442034 INFO AppDaemon: Processing restart for SNIPSMQTT
2019-01-22 20:00:47.451747 INFO AppDaemon: Terminating maths_intent
2019-01-22 20:00:47.474475 INFO AppDaemon: Terminating travel_intent
2019-01-22 20:00:47.489321 INFO AppDaemon: Terminating home_condition_intent
2019-01-22 20:00:47.491397 INFO AppDaemon: Terminating weather_intent
2019-01-22 20:00:47.513065 INFO AppDaemon: Terminating datetime_intent
2019-01-22 20:00:47.520461 INFO AppDaemon: Terminating tell_jokes_intent
2019-01-22 20:00:47.527583 INFO AppDaemon: Terminating retrieve_diabetics_data
2019-01-22 20:00:47.535163 INFO AppDaemon: Terminating shopping_intent
2019-01-22 20:00:47.553119 INFO AppDaemon: Terminating living_room_av
2019-01-22 20:00:47.583376 INFO AppDaemon: Terminating home_state_intent
2019-01-22 20:00:47.590156 INFO AppDaemon: Terminating master_bedroom_av
2019-01-22 20:00:47.606967 INFO AppDaemon: Terminating hallway_av
2019-01-22 20:00:47.609665 INFO AppDaemon: Terminating news_intent
2019-01-22 20:00:47.612519 INFO AppDaemon: Terminating snips_core
2019-01-22 20:00:47.624738 INFO AppDaemon: Terminating search_youtube_app
2019-01-22 20:00:47.636711 INFO AppDaemon: Initializing app search_youtube_app using class SearchYouTube from module search_youtube_app
2019-01-22 20:00:47.742178 INFO AppDaemon: Initializing app snips_core using class SnipsProcess from module snips_core
2019-01-22 20:00:47.768137 INFO AppDaemon: Initializing app news_intent using class ProcessNews from module news_intent
2019-01-22 20:00:47.783550 INFO AppDaemon: Initializing app hallway_av using class AVSystem from module av_system
2019-01-22 20:00:47.797019 INFO AppDaemon: Initializing app master_bedroom_av using class AVSystem from module av_system
2019-01-22 20:00:47.846622 INFO AppDaemon: Initializing app home_state_intent using class ProcessHomeState from module home_state_intent
2019-01-22 20:00:47.860331 INFO AppDaemon: Initializing app living_room_av using class AVSystem from module av_system
2019-01-22 20:00:47.884324 INFO AppDaemon: Initializing app shopping_intent using class ProcessShopList from module shopping_intent
2019-01-22 20:00:47.898216 INFO AppDaemon: Initializing app retrieve_diabetics_data using class RetrieveDiabeticsData from module retrieve_diabetics_data
2019-01-22 20:00:47.910028 INFO AppDaemon: Initializing app tell_jokes_intent using class ProcessJokes from module jokes_tell_intent
2019-01-22 20:00:47.959157 INFO AppDaemon: Initializing app datetime_intent using class ProcessDateTime from module datetime_intent
2019-01-22 20:00:47.975424 INFO AppDaemon: Initializing app weather_intent using class ProcessWeatherQuery from module weather_intent
2019-01-22 20:00:48.000029 INFO AppDaemon: Initializing app home_condition_intent using class ProcessHomeCondition from module home_condition_intent
2019-01-22 20:00:48.014723 INFO AppDaemon: Initializing app travel_intent using class Travel from module travel_intent
2019-01-22 20:00:48.043091 INFO AppDaemon: Initializing app maths_intent using class ProcessMaths from module maths_intent
```

The `snips-core` app has a priority of 10, and that of `search_youtube_app` is on default which is 50. If the priority was followed, the apps with lower priority value should be followed.

**Comment by Odianosen25** [comment]
Going via AD when restarting, it doesn't seem to even be respecting the priority flag when I restart it from command line.

**Comment by Odianosen25** [comment]
Ok I think I have figured it out @acockburn, the issue is that as I got my apps broken down between different `yaml` files, the priority settings don't seem to be respected across the different yaml files.

So if AD could first read all priority settings across the different yaml files, then process it, I believe this can be fixed.

Regards

**Comment by acockburn** [comment]
SHould be fixed now
